src/DateSelector/index.js

```diff
@@ -86,248 +143,150 @@ class DateSelector extends Component {
     }
   }
 
-  handleSelectedPresets () {
-    const { selectedPreset, presets } = this.props
-
-    if (selectedPreset) {
-      const isEqualSelectedPreset = ({ key }) => equals(key, selectedPreset)
-
-      const findItemByKey = pipe(
-        map(({ items }) => items),
-        flatten,
-        find(item => isEqualSelectedPreset(item))
-      )
+  handleOnChange (dates) {
+    const {
+      selectedPreset,
+      onChange,
+      presets,
+    } = this.props
 
-      const foundPreset = findItemByKey(presets)
+    const preset = getPreset(selectedPreset, presets)
 
-      this.handlePresetChange(foundPreset.date(), selectedPreset)
+    if (!selectedPreset || preset.date() === null) {
```

Comment:
You already have `ramda` on this page, why not use an `isNil`?
```suggestion
    if (!selectedPreset || isNil(preset.date())) {
```